frame/system/src/lib.rs

```diff
@@ -164,378 +92,135 @@ pub fn extrinsics_data_root<H: Hash>(xts: Vec<Vec<u8>>) -> H::Output {
-	/// cannot be reaped until this is zero.
-	pub refcount: RefCount,
-	/// The additional data that belongs to this account. Used to store the balance(s) in a lot of
-	/// chains.
-	pub data: AccountData,
-}
-
-/// Stores the `spec_version` and `spec_name` of when the last runtime upgrade
-/// happened.
-#[derive(sp_runtime::RuntimeDebug, Encode, Decode)]
-#[cfg_attr(feature = "std", derive(PartialEq))]
-pub struct LastRuntimeUpgradeInfo {
-	pub spec_version: codec::Compact<u32>,
-	pub spec_name: sp_runtime::RuntimeString,
-}
-
-impl LastRuntimeUpgradeInfo {
-	/// Returns if the runtime was upgraded in comparison of `self` and `current`.
-	///
-	/// Checks if either the `spec_version` increased or the `spec_name` changed.
-	pub fn was_upgraded(&self, current: &sp_version::RuntimeVersion) -> bool {
-		current.spec_version > self.spec_version.0 || current.spec_name != self.spec_name
-	}
-}
-
-impl From<sp_version::RuntimeVersion> for LastRuntimeUpgradeInfo {
-	fn from(version: sp_version::RuntimeVersion) -> Self {
-		Self {
-			spec_version: version.spec_version.into(),
-			spec_name: version.spec_name,
-		}
-	}
-}
-
-decl_storage! {
-	trait Store for Module<T: Config> as System {
-		/// The full account information for a particular account ID.
-		pub Account get(fn account):
-			map hasher(blake2_128_concat) T::AccountId => AccountInfo<T::Index, T::AccountData>;
-
-		/// Total extrinsics count for the current block.
-		ExtrinsicCount: Option<u32>;
-
-		/// The current weight for the block.
-		BlockWeight get(fn block_weight): ConsumedWeight;
-
-		/// Total length (in bytes) for all extrinsics put together, for the current block.
-		AllExtrinsicsLen: Option<u32>;
-
-		/// Map of block numbers to block hashes.
-		pub BlockHash get(fn block_hash) build(|_| vec![(T::BlockNumber::zero(), hash69())]):
-			map hasher(twox_64_concat) T::BlockNumber => T::Hash;
-
-		/// Extrinsics data for the current block (maps an extrinsic's index to its data).
-		ExtrinsicData get(fn extrinsic_data): map hasher(twox_64_concat) u32 => Vec<u8>;
-
-		/// The current block number being processed. Set by `execute_block`.
-		Number get(fn block_number): T::BlockNumber;
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use super::pallet_prelude::*;
+	use crate as frame_system;
+	use frame_support::pallet_prelude::*;
+
+	/// System configuration trait. Implemented by runtime.
+	#[pallet::config]
+	#[pallet::disable_frame_system_supertrait_check]
+	pub trait Config: 'static + Eq + Clone {
+		/// The basic call filter to use in Origin. All origins are built with this filter as base,
+		/// except Root.
+		type BaseCallFilter: Filter<Self::Call>;
+
+		/// Block & extrinsics weights: base values and limits.
+		#[pallet::constant]
+		type BlockWeights: Get<limits::BlockWeights>;
+
+		/// The maximum length of a block (in bytes).
+		type BlockLength: Get<limits::BlockLength>;
```

Comment:
I think we should put it into constant, even if it wasn't
```suggestion
		#[pallet::constant]
		type BlockLength: Get<limits::BlockLength>;
```